crates/stackable-operator/src/product_config_utils.rs

```diff
@@ -1049,6 +1061,140 @@ mod tests {
+                    "file".to_string() => Some(GROUP_CONF_OVERRIDE.to_string()),
+                }
+            }
+        }
+    )]
+    #[case(
+        HashMap::from([
+            ("env".to_string(), ROLE_ENV_OVERRIDE.to_string()),
+        ]),
+        HashMap::from([]),
+        BTreeMap::from([
+            ("cli".to_string(), ROLE_CLI_OVERRIDE.to_string()),
+        ]),
+        BTreeMap::from([]),
+        HashMap::from([
+            ("file".to_string(), HashMap::from([
+                ("file".to_string(), ROLE_CONF_OVERRIDE.to_string())
+            ]))
+        ]),
+        HashMap::from([]),
+        collection ! {
+            ROLE_GROUP.to_string() => collection ! {
+                PropertyNameKind::Env => collection ! {
+                    "env".to_string() => Some(ROLE_ENV_OVERRIDE.to_string()),
+                },
+                PropertyNameKind::Cli => collection ! {
+                    "cli".to_string() => Some(ROLE_CLI_OVERRIDE.to_string()),
+                },
+                PropertyNameKind::File("file".to_string()) => collection ! {
+                    "file".to_string() => Some(ROLE_CONF_OVERRIDE.to_string()),
+                }
+            }
+        }
+    )]
+    #[case(
+        HashMap::from([]),
+        HashMap::from([]),
+        BTreeMap::from([]),
+        BTreeMap::from([]),
+        HashMap::from([]),
+        HashMap::from([]),
+        collection ! {
+            ROLE_GROUP.to_string() => collection ! {
+                PropertyNameKind::Env => collection ! {
+                    "env".to_string() => Some(GROUP_ENV.to_string()),
+                },
+                PropertyNameKind::Cli => collection ! {
+                    "cli".to_string() => Some(GROUP_CLI.to_string()),
+                },
+                PropertyNameKind::File("file".to_string()) => collection ! {
+                    "file".to_string() => Some(GROUP_CONFIG.to_string()),
+                }
+            }
+        }
+    )]
+    fn test_order_in_transform_role_to_config(
+        #[case] role_env_override: HashMap<String, String>,
+        #[case] group_env_override: HashMap<String, String>,
+        #[case] role_cli_override: BTreeMap<String, String>,
+        #[case] group_cli_override: BTreeMap<String, String>,
+        #[case] role_conf_override: HashMap<String, HashMap<String, String>>,
+        #[case] group_conf_override: HashMap<String, HashMap<String, String>>,
+        #[case] expected: HashMap<
+            String,
+            HashMap<PropertyNameKind, BTreeMap<String, Option<String>>>,
+        >,
+    ) {
+        let role: Role<Box<TestConfig>, TestRoleConfig> = Role {
+            config: build_common_config(
+                Some(Box::new(TestConfig {
+                    conf: Some(ROLE_CONFIG.to_string()),
+                    env: Some(ROLE_ENV.to_string()),
+                    cli: Some(ROLE_CLI.to_string()),
+                })),
```

Comment:
suggestion: It seems like this can be replaced with:

```suggestion
                build_test_config(ROLE_CONFIG, ROLE_ENV, ROLE_CLI),
```